powerauth-webflow-authentication-form/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/method/form/controller/FormLoginController.java

```diff
@@ -205,4 +220,56 @@ public UsernamePasswordAuthenticationResponse continueAuthentication(String oper
             return response;
         }
     }
+
+    /**
+     * Get organization detail.
+     * @param request Get organization detail request.
+     * @return Get organization detail response.
+     * @throws AuthStepException Thrown when request is invalid or communication with Next Step fails.
+     */
+    @RequestMapping(value = "/organization", method = RequestMethod.POST)
+    public @ResponseBody OrganizationDetailResponse getOrganizationDetail(@RequestBody OrganizationDetailRequest request) throws AuthStepException {
+        if (request == null || request.getOrganizationId() == null) {
+            throw new AuthStepException("Invalid request in getOrganization", "error.invalidRequest");
+        }
+        String organizationId = request.getOrganizationId();
+        logger.info("Get organization detail started, organization ID: {}", organizationId);
+        final OrganizationDetailResponse response;
+        try {
+            ObjectResponse<GetOrganizationResponse> nsObjectResponse = nextStepClient.getOrganization(organizationId);
+            GetOrganizationResponse nsResponse = nsObjectResponse.getResponseObject();
+            response = organizationConverter.fromNSOrganization(nsResponse);
+        } catch (NextStepServiceException e) {
+            throw new CommunicationFailedException("Organization is not available");
+        }
+        logger.info("Get organization detail succeeded, organization ID: {}", organizationId);
+        return response;
+    }
+
+    /**
+     * Get organizations.
+     * @param request Get organizations request.
+     * @return Get organizations response.
+     * @throws AuthStepException Thrown when request is invalid or communication with Next Step fails.
+     */
+    @RequestMapping(value = "/organization/list", method = RequestMethod.POST)
```

Comment:
I am not sure we should expose organization in such a raw form from the Web Flow component. We should probably create some `./api/auth/form/setup` endpoint with the generified login form setup...

Reply:
Ok, let's change the endpoint.